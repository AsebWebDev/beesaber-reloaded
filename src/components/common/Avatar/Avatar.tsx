import styled from 'styled-components';

import createLocalImageUrl from '@/helper/createLocalImageUrl';
import { parseAvatarUrl } from '@/helper/urlParser';

const Container = styled.img`
  margin-right: 1rem;
  max-width: 2rem;
`;

type Props = {
  avatar?: string;
  playerName: string;
};
const Avatar = ({ avatar, playerName }: Props): JSX.Element => {
  const url =
    avatar === '/images/steam.png' ||
    avatar === '/images/oculus.png' ||
    avatar === undefined
      ? createLocalImageUrl('bee.jpg')
      : parseAvatarUrl(avatar);

  return <Container src={url} alt={`Avatar of player ${playerName}`} />;
};

export default Avatar;
