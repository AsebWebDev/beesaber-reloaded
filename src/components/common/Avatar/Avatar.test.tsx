import { render, screen } from '@testing-library/react';

import exampleBee from '@/testing/testData/exampleBee';

import Avatar from './Avatar';

describe('components/common/Avatar', () => {
  const { avatar, playerName } = exampleBee;

  it('should match snapshot', () => {
    const { container } = render(
      <Avatar avatar={avatar} playerName={playerName} />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it.each`
    avatar           | avatarLink
    ${'not present'} | ${undefined}
    ${'oculus'}      | ${'/images/oculus.png'}
    ${'steam'}       | ${'/images/steam.png'}
    ${'valid/link'}  | ${'valid/link'}
  `(
    'should show avatar placeholder, when avatar is $avatar',
    (avatarLink: string) => {
      render(
        <table>
          <tbody>
            <Avatar avatar={avatarLink} playerName={playerName} />
          </tbody>
        </table>
      );

      const avatarImg = screen.getByRole('img', {
        name: `Avatar of player ${playerName}`,
      }) as HTMLImageElement;

      expect(avatarImg.src).toContain(
        `https://new.scoresaber.com${avatarLink}`
      );
    }
  );
});
