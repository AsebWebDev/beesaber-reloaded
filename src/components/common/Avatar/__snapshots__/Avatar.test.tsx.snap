// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/common/Avatar should match snapshot 1`] = `
.c0 {
  margin-right: 1rem;
  max-width: 2rem;
}

<img
  alt="Avatar of player example Bee"
  class="c0"
  src="https://new.scoresaber.comsomeAvatarString"
/>
`;
